File: orthanc/webhook.py
import orthanc
import json
import requests

# Fixed API URL - use correct port
API_URL = "http://pacs-api:8080/api/orthanc/webhook"

def OnChange(changeType, level, resource):
    if changeType == orthanc.ChangeType.STABLE_STUDY:
        try:
            payload = {
                "ChangeType": "StableStudy",
                "ID": resource,
                "Path": f"/studies/{resource}",
                "ResourceType": "Study",
                "Seq": 0
            }
            
            orthanc.LogWarning(f"Sending webhook for study {resource} to {API_URL}")
            response = requests.post(API_URL, json=payload, timeout=10)
            orthanc.LogWarning(f"Webhook response for study {resource}: {response.status_code}")
            
            if response.status_code == 200:
                orthanc.LogWarning(f"Successfully processed study {resource}")
            else:
                orthanc.LogError(f"Webhook failed with status {response.status_code}: {response.text}")
                
        except Exception as e:
            orthanc.LogError(f"Error sending webhook for study {resource}: {str(e)}")

orthanc.RegisterOnChangeCallback(OnChange)
orthanc.LogWarning("PACS Webhook plugin loaded successfully!")

File: orthanc/webhook.lua
-- PACS Webhook Script (Lua)
-- This script automatically sends new studies to the PACS API

local API_URL = "http://pacs-api:5000/api/orthanc/webhook"

function OnStableStudy(studyId, tags, metadata)
    print("New stable study detected: " .. studyId)

    local payload = {
        ChangeType = "StableStudy",
        ResourceType = "Study",
        ID = studyId,
        Path = "/studies/" .. studyId,
        Seq = 0
    }

    local jsonPayload = DumpJson(payload)

    -- Retry up to 3 times with a short delay
    for attempt = 1, 3 do
        local success, response = pcall(function()
            return HttpPost(API_URL, jsonPayload, {
                ["Content-Type"] = "application/json"
            })
        end)

        if success and response then
            print("Webhook sent successfully for study: " .. studyId .. " (attempt " .. attempt .. ")")
            return
        else
            print("Webhook attempt " .. attempt .. " failed for study: " .. studyId .. ", retrying in 5s...")
            if attempt < 3 then
                os.execute("sleep 5")
            end
        end
    end

    print("All webhook attempts failed for study: " .. studyId)
end

print("PACS Lua Webhook loaded successfully!")
